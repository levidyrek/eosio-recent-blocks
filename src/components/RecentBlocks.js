import React from 'react';
import PropTypes from 'prop-types';
import Block from './Block';
import '../stylesheets/RecentBlocks.css';

function RecentBlocks(props) {
  const { blocks, fetchError, isFetchingBlocks, reloadRecentBlocks } = props;

  const blockElems = blocks.map(block => <Block key={block.id} block={block} />);

  return (
    <div className="RecentBlocks">
      <button type="button" onClick={reloadRecentBlocks} disabled={isFetchingBlocks}>
        Load
      </button>
      {fetchError && <p className="error">Error fetching: {fetchError}</p>}
      <div className="blockList">{blockElems}</div>
    </div>
  );
}

RecentBlocks.propTypes = {
  blocks: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      num_transactions: PropTypes.number.isRequired,
      raw: PropTypes.string.isRequired,
      timestamp: PropTypes.string.isRequired,
    })
  ).isRequired,
  isFetchingBlocks: PropTypes.bool.isRequired,
  fetchError: PropTypes.string,
  reloadRecentBlocks: PropTypes.func.isRequired,
};

RecentBlocks.defaultProps = {
  fetchError: '',
};

export default RecentBlocks;
