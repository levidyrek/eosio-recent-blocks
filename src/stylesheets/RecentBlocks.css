.block {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

/*
* Add a background if the block is hovered, clicked on, or focused
*/
.block.active, .block:hover, .block .header:focus {
  background-color: #ccc;
}

.block .header {
  padding: 18px;
}

/*
* Hide raw block content by default
*/
.block .raw {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}

/*
* Show raw content when it is active
*/
.block.active .raw {
  display: block;
}

.block .raw pre {
  overflow-x: auto;
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
}
