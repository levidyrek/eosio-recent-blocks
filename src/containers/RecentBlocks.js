import { connect } from 'react-redux';
import { reloadRecentBlocks } from '../slices/recentBlocks';
import RecentBlocks from '../components/RecentBlocks';

const mapStateToProps = state => ({
  blocks: state.recentBlocks.blocks,
  isFetchingBlocks: state.recentBlocks.fetching,
});

const mapDispatchToProps = dispatch => ({
  reloadRecentBlocks: () => dispatch(reloadRecentBlocks()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RecentBlocks);
