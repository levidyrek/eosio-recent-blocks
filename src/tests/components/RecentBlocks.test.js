/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RecentBlocks from '../../components/RecentBlocks';

const getBlocks = () => {
  return Array(10)
    .fill()
    .map((_, index) => getBlockData(index + 1));
};

const getBlockData = id => {
  const timestamp = `2019-10-15T10:48:17.00${id}`;
  return {
    id: id.toString(),
    num_transactions: id,
    raw: `{"timestamp": "${timestamp}"}`,
    timestamp,
  };
};

test('Render component', () => {
  const reloadMock = jest.fn();
  const component = renderer.create(
    <RecentBlocks blocks={getBlocks()} isFetchingBlocks={false} reloadRecentBlocks={reloadMock} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Fetching blocks', () => {
  const reloadMock = jest.fn();
  const component = renderer.create(
    <RecentBlocks blocks={getBlocks()} isFetchingBlocks reloadRecentBlocks={reloadMock} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Click load button', () => {
  const reloadMock = jest.fn();
  const component = shallow(
    <RecentBlocks blocks={getBlocks()} isFetchingBlocks reloadRecentBlocks={reloadMock} />
  );

  component.find('button').simulate('click');
  expect(reloadMock.mock.calls.length).toBe(1);
});
