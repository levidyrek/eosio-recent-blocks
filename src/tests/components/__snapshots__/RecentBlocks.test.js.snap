// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch error 1`] = `
<div
  className="RecentBlocks"
>
  <button
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Load
  </button>
  <p
    className="error"
  >
    Error fetching: 
    Network request failed.
  </p>
  <div
    className="blockList"
  >
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          1
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.001
        </h5>
        <h5>
          Transactions: 
          1
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.001"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          2
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.002
        </h5>
        <h5>
          Transactions: 
          2
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.002"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          3
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.003
        </h5>
        <h5>
          Transactions: 
          3
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.003"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          4
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.004
        </h5>
        <h5>
          Transactions: 
          4
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.004"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          5
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.005
        </h5>
        <h5>
          Transactions: 
          5
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.005"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          6
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.006
        </h5>
        <h5>
          Transactions: 
          6
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.006"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          7
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.007
        </h5>
        <h5>
          Transactions: 
          7
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.007"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          8
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.008
        </h5>
        <h5>
          Transactions: 
          8
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.008"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          9
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.009
        </h5>
        <h5>
          Transactions: 
          9
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.009"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          10
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.0010
        </h5>
        <h5>
          Transactions: 
          10
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.0010"}
        </pre>
      </div>
    </div>
  </div>
</div>
`;

exports[`Fetching blocks 1`] = `
<div
  className="RecentBlocks"
>
  <button
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  >
    Load
  </button>
  
  <div
    className="blockList"
  >
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          1
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.001
        </h5>
        <h5>
          Transactions: 
          1
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.001"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          2
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.002
        </h5>
        <h5>
          Transactions: 
          2
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.002"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          3
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.003
        </h5>
        <h5>
          Transactions: 
          3
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.003"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          4
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.004
        </h5>
        <h5>
          Transactions: 
          4
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.004"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          5
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.005
        </h5>
        <h5>
          Transactions: 
          5
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.005"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          6
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.006
        </h5>
        <h5>
          Transactions: 
          6
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.006"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          7
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.007
        </h5>
        <h5>
          Transactions: 
          7
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.007"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          8
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.008
        </h5>
        <h5>
          Transactions: 
          8
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.008"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          9
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.009
        </h5>
        <h5>
          Transactions: 
          9
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.009"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          10
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.0010
        </h5>
        <h5>
          Transactions: 
          10
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.0010"}
        </pre>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render component 1`] = `
<div
  className="RecentBlocks"
>
  <button
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Load
  </button>
  
  <div
    className="blockList"
  >
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          1
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.001
        </h5>
        <h5>
          Transactions: 
          1
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.001"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          2
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.002
        </h5>
        <h5>
          Transactions: 
          2
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.002"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          3
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.003
        </h5>
        <h5>
          Transactions: 
          3
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.003"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          4
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.004
        </h5>
        <h5>
          Transactions: 
          4
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.004"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          5
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.005
        </h5>
        <h5>
          Transactions: 
          5
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.005"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          6
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.006
        </h5>
        <h5>
          Transactions: 
          6
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.006"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          7
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.007
        </h5>
        <h5>
          Transactions: 
          7
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.007"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          8
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.008
        </h5>
        <h5>
          Transactions: 
          8
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.008"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          9
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.009
        </h5>
        <h5>
          Transactions: 
          9
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.009"}
        </pre>
      </div>
    </div>
    <div
      className="block "
    >
      <div
        className="header"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <h3>
          ID: 
          10
        </h3>
        <h5>
          Timestamp: 
          2019-10-15T10:48:17.0010
        </h5>
        <h5>
          Transactions: 
          10
        </h5>
      </div>
      <div
        className="raw"
      >
        <h5>
          Raw content
        </h5>
        <pre>
          {"timestamp": "2019-10-15T10:48:17.0010"}
        </pre>
      </div>
    </div>
  </div>
</div>
`;
